src/DDBFieldPlugin.cc

```diff
@@ -0,0 +1,43 @@
+/**
+ *  @file   DDMarlinPandora/src/DDBFieldPlugin.cc
+ * 
+ *  @brief  Implementation of the geometry creator class.
+ * 
+ *  $Log: $
+ */
+
+#include "DDBFieldPlugin.h"
+
+#include "Helpers/XmlHelper.h"
+
+
+DDBFieldPlugin::DDBFieldPlugin(const dd4hep::Detector &detector) :
+    m_field(detector.field())
+{
+    /* nop */
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+float DDBFieldPlugin::GetBField(const pandora::CartesianVector &positionVector) const
+{
+    double bfield[3] = {0.};
+    m_field.magneticField({positionVector.GetX(), positionVector.GetY(), positionVector.GetZ()}, bfield);
```

Comment:
I think you need to do unit conversion for X,Y,Z

Reply:
Done. Is this correct ?